### ISSUE TITLE
Stats: Invalid requests for referrers for site `0` when viewing `/stats/year` page on Safari

### ISSUE THREAD

**Issue by mattsherman** [Automattic/wp-calypso#2854]
When viewing a stats URL such as:

`https://wordpress.com/stats/year/example.com`

in Safari, there are two requests to:

`https://public-api.wordpress.com/rest/v1.1/sites/0/stats/referrers`

that result in 404s.

Then, the correct `referrers` API call is made.

These invalid `referrers` calls only happen in Safari. Chrome and Firefox do not exhibit this.

I tried clearing my cache and local storage and still observe the same behavior.

Expected behavior:
- One valid call to `referrers` endpoint for correct site ID is made

Observed behavior:
- Two invalid calls to `referrers` endpoint for site `0` is made
- One valid call to `referrers` endpoint for correct site ID is made

**Comment by jblz** [comment]
This looks like a duplicate of #2400. Is this a distinct bug?

**Comment by mattsherman** [comment]
I was not aware of that issue, but after looking at it, I believe this is a distinct bug (though perhaps related), because of the following:

1. It was happening before I cleared my localstorage
2. It continues to happen even after my localstorage is populated again -- I can't figure out a way to stop this in Safari
3. I observe it happening only in Safari

**Comment by timmyc** [comment]
Just tested this out after #2459 has been merged in and I can not recreate it anymore.  Closing this one out.
